src/Sylius/Component/Product/Generator/ProductVariantGenerator.php

```diff
@@ -56,6 +56,10 @@ public function generate(ProductInterface $product): void
             }
         }
 
+        if (empty($optionSet)) {
+            throw new \InvalidArgumentException('Cannot generate variants for a product without options');
```

Comment:
`\InvalidArgumentException` is ok right now, but we can think about something more sophisticated, like `VariantWithNoOptionsValuesException`